tools/ci-build/publisher/src/subcommand/claim_crate_names.rs

```diff
@@ -79,20 +77,11 @@ async fn discover_publishable_crate_names(repository_root: &Path) -> anyhow::Res
         fs: Fs,
         path: PathBuf,
     ) -> anyhow::Result<HashSet<String>> {
-        let manifest_paths = discover_package_manifests(path).await?;
+        let packages = discover_packages(fs, path).await?;
         let mut publishable_package_names = HashSet::new();
-        for manifest_path in manifest_paths {
-            let contents: Vec<u8> = fs.read_file(&manifest_path).await?;
-            let manifest = Manifest::from_slice(&contents).with_context(|| {
-                format!("failed to load package manifest for {:?}", manifest_path)
-            })?;
-            let package = manifest
-                .package
-                .ok_or(Error::InvalidManifest(manifest_path))
-                .context("crate manifest doesn't have a `[package]` section")?;
-            let name = package.name;
-            if let cargo_toml::Publish::Flag(true) = package.publish {
-                publishable_package_names.insert(name);
+        for package in packages {
+            if let Publish::Allowed = package.publish {
```

Comment:
Mostly for my understanding. Is it ok to change `Publish::Flag(true)` (previous code) to `Publish::Allowed`?

Reply:
They're different types. `Publish::Flag` is part of the `cargo_toml` library, whereas `Publish::Allowed` is from `crate::package`. This change made it use our existing package code rather than directly parsing the manifest and interacting with that parse result.